streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -41,13 +45,19 @@
 import static org.apache.kafka.test.MockStateRestoreListener.RESTORE_BATCH;
 import static org.apache.kafka.test.MockStateRestoreListener.RESTORE_END;
 import static org.apache.kafka.test.MockStateRestoreListener.RESTORE_START;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsEqual.equalTo;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
+@RunWith(EasyMockRunner.class)
 public class StoreChangelogReaderTest {
 
+    @Mock(type = MockType.NICE)
```

Comment:
I guess we should add a test in here that triggers the `TaskMigratedException` ?

Reply:
Added test and discovered a bug in `StoreChangelogReader#processNext` -- we should always return `consumer.position(restorer.partition())`, otherwise, we don't detect if there is more data than expected. \cc @guozhangwang @bbejeck

Reply:
really? we have the position already from either the last record offset or from the previous call to `consumer.position(...)`